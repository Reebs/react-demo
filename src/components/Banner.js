import React, {Component} from 'react';

export default class Banner extends Component {

    componentWillMount()
    {
        // console.log("Banner.componentWillMount")
    }

    render() {
        console.log("Banner.render")
        return  <div>
                    <h1>{this.props.banner}</h1>
                </div>
    }

    componentDidMount() {
        // console.log("Banner.componentDidMount")
    }
}
