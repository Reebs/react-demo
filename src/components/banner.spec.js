import React from 'react';
import renderer from 'react-test-renderer'
import Banner from "./Banner";

test('banner shows up when rendered', () => {
  
  const banner = renderer.create(<Banner banner='hello world!' />)
  let tree = banner.toJSON();
  expect(tree).toMatchSnapshot();
})
