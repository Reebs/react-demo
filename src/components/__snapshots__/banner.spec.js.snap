// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner shows up when rendered 1`] = `
<h1>
  hello world!
</h1>
`;
